span.go

```diff
@@ -145,3 +150,13 @@ func CanonicalizeTraceAttributeKey(key string) (string, bool) {
 	}
 	return strings.ToLower(key), true
 }
+
+// InjectSpan is a helper that takes some of the stuttering out of the common
+// Span.WireInjectorForEncoding calling pattern.
+func InjectSpan(sp Span, enc WireEncoding, carrier ...interface{}) error {
```

Comment:
on the other hand, the more parameters you have in one method the harder it is for the new-comer to understand what is actually going on. Especially when it works one way here but another way over there. 

That and it's more porcelain api that can get added once the plumbing is working.